sbt-coursier/src/main/scala-2.10/coursier/ToSbt.scala

```diff
@@ -1,90 +1,103 @@
 package coursier
 
 import java.util.GregorianCalendar
+import java.util.concurrent.ConcurrentHashMap
 
 import coursier.maven.MavenSource
-
 import sbt._
 
 object ToSbt {
 
-  def moduleId(dependency: Dependency): sbt.ModuleID =
+  private def caching[K, V](f: K => V): K => V = {
+
+    val cache = new ConcurrentHashMap[K, V]
```

Comment:
Thanks for the hint. There are people raising good concerns about `scala.collection.concurrent.TrieMap`, like [here](http://42ways.de/blog/2015/02/21/scala-pitfalls-concurrent-hashmap/) or [there](http://stackoverflow.com/questions/29499381/what-is-a-triemap-and-what-is-its-advantages-disadvantages-compared-to-a-hashmap/38700133#38700133) (which should explain why it's not that known). The risk of using it wouldn't be that high here though, I'll probably update the PR with it...